src/screens/NanoContract/NanoContractRegisterScreen.js

```diff
@@ -0,0 +1,323 @@
+  };
+
+  const navigatesToNanoContractTransactions = () => {
+    dispatch(nanoContractRegisterReady());
+    navigation.replace('NanoContractDetailsScreen', { ncId: formModel.ncId });
+  };
+
+  useEffect(() => {
+    dispatch(firstAddressRequest());
+  }, []);
+
+  const hasFirstAddressFailed = () => error;
+  const isFirstAddressLoading = () => !error && !address;
+  const hasFirstAddressLoaded = () => !error && address;
+
+  return (
+    <Wrapper>
+      <NavigationHeader navigation={navigation} />
+
+      {hasSucceeded(registerState.registerStatus)
+        && (
+          <FeedbackModal
+            icon={(<Image source={checkIcon} style={styles.feedbackModalIcon} resizeMode='contain' />)}
+            text={feedbackSucceedText}
+            onDismiss={handleFeedbackModalDismiss}
+            action={(<NewHathorButton discrete title={t`See contract`} onPress={navigatesToNanoContractTransactions} />)}
+          />
+        )}
+
+      {hasFailed(registerState.registerStatus)
+        && (
+          <FeedbackModal
+            icon={(<Image source={errorIcon} style={styles.feedbackModalIcon} resizeMode='contain' />)}
+            text={registerState.registerFailureMessage}
+            onDismiss={handleFeedbackModalDismiss}
+            action={(<NewHathorButton discrete title={t`Try again`} onPress={handleSubmit} />)}
+          />
+        )}
+
+      {hasFirstAddressFailed()
+        && (
+        <FeedbackContent
+          icon={(<Image source={errorIcon} style={styles.feedbackContentIcon} resizeMode='contain' />)}
+          title={t`Load First Addresses Error`}
+          message={error}
+        />
+        )}
+
+      {isFirstAddressLoading()
+        && (
+        <FeedbackContent
+          title={t`Loading`}
+          message={t`Loading first wallet address.`}
+        />
+        )}
+
+      {hasFirstAddressLoaded()
+        && (
+        <ContentWrapper>
+          <SimpleInput
+            containerStyle={styles.input}
+            label={t`Nano Contract ID`}
+            autoFocus
+            onChangeText={handleInputChange('ncId')}
+            error={invalidModel.ncId}
+            value={formModel.ncId}
+          />
+          <View style={styles.selectionContainer}>
+            <FieldContainer>
+              <TextLabel pb8 bold>{t`Wallet Address`}</TextLabel>
+              <TextValue>{address}</TextValue>
+            </FieldContainer>
+          </View>
+          <View style={styles.infoContainer}>
+            <View style={styles.infoIcon}>
+              <CircleInfoIcon size={20} color='hsla(203, 100%, 25%, 1)' />
+            </View>
+            <View style={styles.infoContent}>
+              <Text style={styles.text}>
+                {t`If you want to change the wallet address, you will be able to do`}
+                <Text style={styles.bold}>
+                  {' '}{t`after the contract is registered.`}
+                </Text>
+              </Text>
+            </View>
```

Comment:
I think this view should have a padding right as well (at least 4). The text doesn't seem to fit well, because it's really close to the border. 

![image](https://github.com/HathorNetwork/hathor-wallet-mobile/assets/3298774/bbd6fc7f-ad83-47cc-8fe8-f53027b52b7c)